Dockerfile

```
@@ -1,38 +1,63 @@
 FROM ubuntu:18.04
 
-RUN mkdir /data
+RUN mkdir /data /package
 
-ENV BBLFSH_HOSTNAME=tmexp_bblfshd \
-  BBLFSH_PORT=9432 \
-  GITBASE_HOSTNAME=tmexp_gitbase \
-  GITBASE_PORT=3306 \
-  GITBASE_USERNAME=root \
-  GITBASE_PASSWORD=
+ARG DEBIAN_FRONTEND=noninteractive 
+
+ENV BBLFSH_HOSTNAME tmexp_bblfshd
+ENV BBLFSH_PORT 9432
+ENV GITBASE_HOSTNAME tmexp_gitbase
+ENV GITBASE_PORT 3306
+ENV GITBASE_USERNAME root
+ENV GITBASE_PASSWORD ""
+ENV ARTM_SHARED_LIBRARY /usr/local/lib/libartm.so
+
+SHELL ["/bin/bash", "-o", "pipefail", "-c"]
 
 RUN apt-get update \
-  && apt-get install -y dialog apt-utils curl \
+  && apt-get install -y --no-install-recommends \
+  apt-utils \
+  ca-certificates \
+  curl \
+  locales \
+  && echo "en_US.UTF-8 UTF-8" > /etc/locale.gen \
+  && locale-gen \
   && apt-get clean \
   && rm -rf /var/lib/apt/lists/*
 
-RUN curl -sSL https://get.docker.com/ | sh 
-
-RUN apt-get update \
-  && apt-get install -y python3-pip python3-dev \
-  && cd /usr/local/bin \
-  && ln -s /usr/bin/python3 python \
-  && apt-get remove -y .*-doc .*-man >/dev/null \
-  && apt-get autoremove --purge -y \ 
+RUN curl -sSL https://get.docker.com/ | sh \
+  && apt-get update \
+  && apt-get install -y --no-install-recommends \
+  build-essential \
+  cmake \
+  git \
+  libboost-all-dev \
+  make \
+  python3-dev \
+  python3-pip \
+  && ln -s /usr/bin/python3 /usr/local/bin/python \
+  && pip3 install numpy scipy pandas protobuf tqdm wheel \
+  && git clone --branch v0.10.0 https://github.com/bigartm/bigartm.git /opt/bigartm \
+  && cd /opt/bigartm \
+  && git reset --hard 3c0cbc4ac08cae37f00754815c105fde930ff379 \
+  && mkdir build \
+  && cd build \
+  && cmake -DINSTALL_PYTHON_PACKAGE=ON -DPYTHON=python3 .. \
+  && make \
```

```suggestion
  && make -j$(getconf _NPROCESSORS_ONLN) \
```
and your build becomes fast again 😄 `getconf` is from `libc-bin`

---

somewhat fast since installing liboost-all-dev takes forever :D thanks for the tip!

---

You don't need to install it btw. Boost is spread into many separate feature packages, so you should install only those which are required.

---

yeah but they don't specify which ones are required and it will take forever to find out. If you have a trick let me know though!

---

the list of required libs 
```
  libboost-dev \
  libboost-thread-dev \
  libboost-program-options-dev \
  libboost-date-time-dev \
  libboost-filesystem-dev \
  libboost-iostreams-dev \
  libboost-system-dev \
  libboost-chrono-dev \
  libboost-timer-dev \ 
```